cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -1261,6 +1268,28 @@ private String getHiveDatabase(String namespace) {
     return sessionConf;
   }
 
+  /**
+   * Updates the token store to be used for the hive job, based upon the Explore container's credentials.
+   * This is because twill doesn't update the container_tokens on upon token refresh.
+   * See: https://issues.apache.org/jira/browse/TWILL-170
+   */
+  private void updateTokenStore() throws IOException, ExploreException {
+    File credentialsFile = new File(System.getenv(UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION));
```

Comment:
Should guard against in case the env variable is missing. Maybe just ignore the update (and log) if there is no such env variable.